packages/neo-one-client/src/user/LocalUserAccountProvider.ts

```diff
@@ -823,6 +841,172 @@ export class LocalUserAccountProvider<TKeyStore extends KeyStore, TProvider exte
     );
   }
 
+  private async consolidate({
+    inputs,
+    transactionUnsignedIn,
+    from,
+    monitor,
+    byteLimit,
+  }: {
+    readonly transactionUnsignedIn: TransactionModel;
+    readonly inputs: ReadonlyArray<UnspentOutput>;
+    readonly from: UserAccountID;
+    readonly monitor?: Monitor;
+    readonly byteLimit: number;
+  }): Promise<TransactionModel> {
+    const newInputSize = 68;
+    const newOutputSize = 120;
+    const messageSize = transactionUnsignedIn.serializeUnsigned().toString('hex').length;
+
+    if (byteLimit - messageSize < 0) {
+      throw new TransactionSizeExceedsLimitError(messageSize, byteLimit);
+    }
+    if (byteLimit - messageSize - newInputSize < 0) {
+      return transactionUnsignedIn;
+    }
+
+    const inputMap = (): { [key: string]: ReadonlyArray<UnspentOutput> } => ({});
+    const inputArray = (): ReadonlyArray<InputModel> => [];
+    const outputArray = (): ReadonlyArray<OutputModel> => [];
+    const getMessageSize = (numInputs: number, numOutputs: number) =>
+      messageSize + numInputs * newInputSize + numOutputs * newOutputSize;
+
+    const updateOutput = (output: OutputModel, amount: BigNumber): OutputModel =>
+      new OutputModel({
+        asset: output.asset,
+        value: output.value.add(clientUtils.bigNumberToBN(amount, 8)),
+        address: output.address,
+      });
+
+    const consoladatableUnspentsIn = await this.provider.getUnspentOutputs(from.network, from.address, monitor);
+
+    const looseAssets = consoladatableUnspentsIn
```

Comment:
I would call this `assetToUnspentOutputs`. I have no idea what `looseAssets` means, but `assetToUnspentOutputs` means exactly what it says, it's a map from asset to unspent outputs.